memgpt/schemas/memgpt_message.py

```diff
@@ -32,6 +35,20 @@ class FunctionCall(BaseModel):
     arguments: str
 
 
+class FunctionCallDelta(BaseModel):
```

Comment:
@sarahwooders I didn't have to make a `Delta` / `Chunk` specific model for `InnerMonologue` since `InnerMonologue` has chunk support built in - `InnerMonologue.arguments` can just be partial pieces:

```
InnerMonologue.arguments = "hello there"
```

to

```
InnerMonologue.arguments = "hello "
InnerMonologue.arguments = "there"
```

However `FunctionCall` is problematic since at least with OpenAI API the stream back usually starts with just `name`, then chunks of the `arguments`:

```
FunctionCall.name: "send_message"
FunctionCall.arguments: "\{\ 'content': ...
```

to

```
FunctionCall.name: "send_message"
FunctionCall.arguments: "\{\ "
FunctionCall.arguments: "'content:'"
...
```

So we need a new Pydantic model that supports optional attributes when name is null or arguments is null (technically you should never have the case where both are null, but not sure how you set that up in Pydantic + probably not worth the hassle).